File: src/main/java/de/silencio/activecraftcore/guicreator/GuiConfirmEvent.java
package de.silencio.activecraftcore.guicreator;

import de.silencio.activecraftcore.events.ActiveCraftEvent;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;

@Data
@EqualsAndHashCode(callSuper = false)
public class GuiConfirmEvent extends ActiveCraftEvent {

    private boolean cancelled;
    private Gui gui;
    private @NonNull Inventory clickedInventory;
    private @NonNull String identifier;
    private @NonNull Player player;

    public Gui getGui() {
        return gui != null ? gui : (gui = Gui.ofInventory(clickedInventory));
    }
}

File: src/main/java/org/activecraft/activecraftcore/guicreator/GuiConfirmEvent.kt
package org.activecraft.activecraftcore.guicreator

import org.activecraft.activecraftcore.events.CancellableActiveCraftEvent
import org.activecraft.activecraftcore.guicreator.Gui.Companion.ofInventory
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory

class GuiConfirmEvent(val clickedInventory: Inventory, val identifier: String, val player: Player) : CancellableActiveCraftEvent() {
    var gui: Gui? = null
        get() = if (field != null) field else ofInventory(clickedInventory).also { field = it }
        private set
}
